File: bench/fac.go
package main

import (
	"fmt"
	"math/big"
)

const target = 50000

func main() {
	f := new(big.Int)
	c := new(big.Int)
	one := new(big.Int)

	f.SetInt64(1)
	c.SetInt64(1)
	one.SetInt64(1)

	for i := 0; i < target; i++ {
		f.Mul(f, c)
		c.Add(c, one)
	}

	fmt.Printf("%s", f.String())
}

File: bench/fac.zig
const std = @import("std");
const BigInt = @import("../bigint.zig").BigInt;

const target = 50000;

pub fn main() !void {
    var stdout_file = try std.io.getStdOut();
    var allocator = std.heap.c_allocator;

    var f = try BigInt.initSet(allocator, 1);
    var c = try BigInt.initSet(allocator, 1);
    var one = try BigInt.initSet(allocator, 1);

    var i: usize = 0;
    while (i < target) : (i += 1) {
        try f.mul(&f, &c);
        try c.add(&c, &one);
    }

    const r = try f.toString(allocator, 16);
    try stdout_file.write(r);
}
